src/query_plan/operation.rs

```diff
@@ -14,10 +14,24 @@ use apollo_compiler::{name, Node};
 use indexmap::{IndexMap, IndexSet};
 use linked_hash_map::{Entry, LinkedHashMap};
 use std::ops::Deref;
-use std::sync::Arc;
+use std::sync::{atomic, Arc};
 
 const TYPENAME_FIELD: Name = name!("__typename");
 
+// Global storage for the counter used to uniquely identify selections
+static NEXT_ID: atomic::AtomicUsize = atomic::AtomicUsize::new(0);
```

Comment:
I think this needs to start at `1` instead of `0`, since we use `label` of `0` below to indicate that there is no defer usage (and `fetch_add()` returns the value prior to the add).